src/matching/Match_patterns.ml

```diff
@@ -258,6 +258,16 @@ let check2 ~hook mvar_context range_filter (config, equivs) rules
            | Fld pattern -> Common.push (pattern, rule) fld_rules
            | Flds pattern -> Common.push (pattern, rule) flds_rules
            | Partial pattern -> Common.push (pattern, rule) partial_rules
+           (* If we receive an incoming Name pattern, we want to use it
+              as a name only if it's not a metavariable because then it
+              should be matched against more nodes.
+              So we sort based on if its a metavariable.
+           *)
+           (* THINK: should we do this for named ellipses also? *)
+           | Name (Id ((s, _), _) as name) when Metavariable.is_metavar_name s
```

Comment:
Shouldn't we perhaps convert this earlier to an `E (Id ...)` like in `Parse_rust_tree_sitter`?

Reply:
My understanding is that in the other languages a standalone `$X` is just parsed as `E (Id ...)`, so that's the "contract" and `Parse_rust_tree_sitter` breaks it.

Reply:
Hmm, I was thinking about it being useful for names to remain name patterns. But I realize you're saying just to convert the metavariable case, which I think is reasonable. I think we can actually do this in `Parse_pattern`, instead, you're right.